src/DynamicData/Cache/ObservableCacheEx.BindSort.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2011-2023 Roland Pheasant. All rights reserved.
+// Roland Pheasant licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for full license information.
+
+using System.Collections.ObjectModel;
+using DynamicData.Binding;
+
+namespace DynamicData;
+
+/// <summary>
+/// ObservableCache extensions for BindAndSort.
+/// </summary>
+public static partial class ObservableCacheEx
+{
+    /// <summary>
+    /// Bind sorted data to the specified readonly observable collection.
+    /// </summary>
+    /// <typeparam name="TObject">The type of the object.</typeparam>
+    /// <typeparam name="TKey">The type of the key.</typeparam>
+    /// <param name="source">The source.</param>
+    /// <param name="targetList">The list to bind to.</param>
+    /// <param name="comparer">The comparer to order the resulting dataset.</param>
+    /// <returns>An observable which will emit change sets.</returns>
+    public static IObservable<IChangeSet<TObject, TKey>> BindAndSort<TObject, TKey>(
```

Comment:
I think it makes more sense to consider this a "terminating" operator, I.E. have it perform a subscription and return `IDisposable`.

The big code smell in my mind, with leaving subscription management up to the downstream consumer, is that the operator will break if the consumer makes multiple downstream subscription. The operator will end up running *for each subscription* and trying to apply every changeset to the target list multiple times.

If a consumer NEEDS to be able to listen to downstream changes, I think that really necessitates that they split `.BindAndSort()` into separate `.Sort()` and `.Bind()` operators, so that they can share and publish the `.Sort()` downstream, for both` .Bind()` and whatever else they need.

Reply:
In the very earliest version of Dynamic Data I implemented terminating operators.  But they are problematic because:

1. I often end up adding stuff after Bind() in real world apps.
2. If we were to force users to use Publish(), then it's more code for everyone without an real benefit.
3. It's better to let the consumer decide what is terminating.

Besides once SortAndBind has settled I will make Sort obsolete (for cache). Sort is a terrible operator and I have always been somewhat ashamed of how bad it's performance is and dearly want to get rid of if.  My proposal is to complete SortAndBind asap, then in the next major release mark Sort as obsolete then remove it entirely in the following major release.

Reply:
Standard practice is 2 major releases before removing obsolete APIs.

Also, I would kinda like a chance to see if we (or just I) can rework `.Sort()` for caches into something reasonable, before comitting to dropping it. It seems far too limiting to me to say "you can only sort as the very last step", surely there's a way to make it not terrible. Maybe we can workshop it in DD2 and then pull what works into DD.

Reply:
Regarding terminating operators, surely it is something to be considered for Dynamic Data 2 and not the current version.  

For context I have historically done stuff like this:

```cs
   mySource
      .Sort(...)
      .Bind(....)
      .Subscribe(_> 'Do some ui related action which requires the bound list to be updated first' )
```

My preference is not to have them but I am willing to be consider it - you make a good point about subscribing twice, but in practice I don't think anyone has ever complained about that being an issue.

Honestly, do not bother with Sort. I spent more time on it, and had more trouble than on any other operator.  Also the sorted changeset is fatally flawed as it need to maintain a copy of sorted state after ever change, which is transmitted on the sorted change set. This leads to huge memory allocations and the CPU goes crazy, and for what?  The only operators which use sort are Bind, Virtualise and Page.   Without sort it would mean we could also get rid of this pesky sorted change set.  Also without sort in the cache we do not need index on the changes.